src/main/scala/io/iohk/ethereum/blockchain/sync/fast/PivotBlockSelector.scala

```diff
@@ -63,10 +63,13 @@ class PivotBlockSelector(
       log.info(
         "Cannot pick pivot block. Need at least {} peers, but there are only {} which meet the criteria ({} all available at the moment). Retrying in {}",
         minPeersToChoosePivotBlock,
-        correctPeers.size,
-        peersToDownloadFrom.size,
+        eligiblePeers.size,
+        allPeers.size,
         startRetryInterval
       )
+      if (election.isEnoughVoters(minPeersToChoosePivotBlock + 1)) { // TODO when we should start blacklisting best peers?
+        blacklist(eligiblePeers.head.id, blacklistDuration, "The best peer has too big best block number, blacklisting")
```

Comment:
Maybe you can add bestPeer in the case class ElectionDetails. Or you can change the name eligiblePeers by eligiblePeersSortedByBestNumber. I think that by this way, it is more understandable that the list “eligiblepeers” was sorted, then you try to get the best peer “eligiblepeers.head.id”.